src/Database/Kioku/Internal/TrieIndex.hs

```diff
@@ -23,22 +24,37 @@ import            System.IO
 import            Database.Kioku.Internal.Buffer
 import            Database.Kioku.Memorizable
 
-
 trieLookup :: BS.ByteString -> TrieIndex -> [Int]
 trieLookup key = maybe [] trieRootElems . lookupSubtrie key
 
 trieMatch :: BS.ByteString -> TrieIndex -> [Int]
 trieMatch prefix = maybe [] trieElems . lookupSubtrie prefix
 
+trieAllHitsAlong :: BS.ByteString -> TrieIndex -> [Int]
+trieAllHitsAlong "" _ = []
+trieAllHitsAlong key trie =
+  let (p,ath) = BS.splitAt 1 key
+  in concat $ go p ath trie
+    where
+      go prefix remaining trie' =
+        let (p', ath') = BS.splitAt 1 remaining
+        in case lookupSubtrie prefix trie' of
+             Nothing -> if BS.null remaining
+                          then []
+                          else go p' ath' trie'
```

Comment:
If there's no subtrie at `prefix`, is it actually correct to then lookup `p'` (which would be the *next* character *after* prefix in the origin query, right?)? Wouldn't we simply stop as soon as there's no subtrie?